code/game/objects/items/devices/pcmc.dm

```diff
@@ -0,0 +1,258 @@
+// A fun toy for Paramedics
+
+/obj/item/device/pcmc
+	name = "Portable Crew Monitoring Computer"
+	desc = "A breakthrough of technology. Easily shows all detected suit sensors, and can filter to only injuries."
+	icon = 'icons/obj/pda.dmi'
+	icon_state = "pcmc"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/pcmc.dmi', "right_hand" = 'icons/mob/in-hand/right/pcmc.dmi')
+	item_state = "pcmc"
+	w_class = W_CLASS_MEDIUM
+	flags = FPRINT
+	slot_flags = SLOT_BELT
+	mech_flags = MECH_SCAN_FAIL
+	force = 8
+	throwforce = 15
+	throw_range = 4
+	attack_verb = list("bricks")
+	var/emped = FALSE
+	var/autorefreshing = FALSE
+	var/transmitting = FALSE
+	var/injuryonly = FALSE
+	var/fullmode = TRUE
+
+/obj/item/device/pcmc/New()
+	..()
+	update_icon()
+
+/obj/item/device/pcmc/Destroy()
+	..()
+
+/obj/item/device/pcmc/update_icon()
+	overlays.Cut()
+	if(emped)
+		overlays += image(icon, "pcmcemp")
+		return
+	if(transmitting)
+		overlays += image(icon, "pcmcon")
+
+/obj/item/device/pcmc/emp_act(severity)
+	emped = TRUE
+	transmitting = FALSE
+	set_light(0)
+	update_icon()
+	SStgui.update_uis(src)
+	spawn(120 SECONDS)
+		emped = FALSE
+		update_icon()
+		SStgui.update_uis(src)
+
+/obj/item/device/pcmc/attack_self(mob/user)
+	tgui_interact(user)
+
+/obj/item/device/pcmc/examine(mob/user)
+	if(Adjacent(user) || isobserver(user))
+		attack_self(user)
+	else
+		..()
+
+/obj/item/device/pcmc/proc/get_location_name()
+	var/turf/device_turf = get_turf(src)
+	var/area/device_area = get_area(src)
+	if (emped)
+		return "ERROR"
+	else if(!device_turf || !device_area)
+		return "UNKNOWN"
+	else if(device_turf.z > WORLD_X_OFFSET.len)
+		return "[format_text(device_area.name)] (UNKNOWN, UNKNOWN, UNKNOWN)"
+	else
+		return "[format_text(device_area.name)] ([device_turf.x-WORLD_X_OFFSET[device_turf.z]], [device_turf.y-WORLD_Y_OFFSET[device_turf.z]], [device_turf.z])"
+
+
+/obj/item/device/pcmc/proc/get_crew()
+	//looping though carbons
+	var/list/crewlist = list()
+	for(var/mob/living/carbon/human/H in mob_list)
+
+		if(H.iscorpse)
+			continue
+		if(H.z == 2)
+			continue
+
+		var/name
+		var/assignment
+		var/life_status
+		var/list/damage = list()
+		damage["oxygen"] = ""
+		damage["toxin"] = ""
+		damage["burn"] = ""
+		damage["brute"] = ""
+		var/player_area
+		var/see_x = "?"
+		var/see_y = "?"
+		var/see_z = "?"
+
+		// z == 0 means mob is inside object, check is they are wearing a uniform
+		if(istype(H.w_uniform, /obj/item/clothing/under))
+			var/obj/item/clothing/under/U = H.w_uniform
+
+			if (U.has_sensor && U.sensor_mode)
+				var/turf/pos = H.z == 0 || U.sensor_mode == 3 ? get_turf(H) : null
```

Comment:
`sensor_mode` should probably be a define but that's not coded yet. You could bundle it with this PR.

Reply:
"Adds PCMC for paramedics and refactors suit sensor code to use defines" yeah no, that's a separate PR. It isn't hard to change over to defines, but I don't feel it fits in a PR that might get reverted if people don't want it. Even I'm saying that this specific PR is technically experimental. If this gets merged, it'll be easy to waltz on in and change the 0 to SUITSENSORS_OFF, 1 to SUITSENSORS_BINARY, 2 to SUITSENSORS_VITALS, 3 to SUITSENSORS_FULL, change that in clothing.dm, cmc.dm, and the few other places that use them, throw a define in, and be done with it.